### PR TITLE
[CI] upgrade CI pytorch version

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
Upgrading CI's PyTorch version to 1.12.

### CODE REVIEW

File: docker/Dockerfile.ci_gpu [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # CI docker GPU env
-FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04
+FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04
 
 ENV TZ=US
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
+RUN apt-key del 3bf863cc
```
Comment (Rhett-Ying) [inline docker/Dockerfile.ci_gpu:7]: this looks a bit weird. could you add a comment to clarify it? It aims to fix nv image issue?

File: docker/Dockerfile.ci_gpu_220816 [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# CI docker GPU env
```
Comment (Rhett-Ying) [inline docker/Dockerfile.ci_gpu_220816:1]: this is used for tensorflow and mxnet? could we keep only one docker for gpu tests?

Comment (BarclayII) [reply docker/Dockerfile.ci_gpu_220816:1]: Somehow Tensorflow crashes with CUDA 10.2 and I wasn't able to make it work.

File: docker/Dockerfile.ci_gpu (hunk 1 above)
Comment (BarclayII) [reply docker/Dockerfile.ci_gpu:8]: Added a comment.

File: docker/Dockerfile.ci_gpu_220816 (hunk 2 above)
Comment (Rhett-Ying) [reply docker/Dockerfile.ci_gpu_220816:1]: all right. let's keep it

**Review by Rhett-Ying — APPROVED** [review]
LGTM

**Pull request merged**